### ISSUE TITLE
Family Strokes Not Matching

### ISSUE THREAD

**Issue by HWatBobby** [PAhelper/PhoenixAdult.bundle#510]
Unable to match Family Strokes scenes since latest scene released. Doesn't look like URL changed from what is in PASearchSites. Filename is FamilyStrokes - 2019-12-26 - 1079 - Christmas Family Orgy.mp4. It is trying to link to https://www.familystrokes.com/scenes/1079-Christmas-Family-Orgy instead of https://www.familystrokes.com/scenes/1079/1/Christmas-Family-Orgy. I'm not sure how to grab two elements from a file name to insert into URL for the search.

Something like ```PAsearchSites.getSearchSearchURL(searchSiteID) + sceneID + "/1/" + sceneTitle```

Log output:
```
2019-12-27 03:55:38,560 (150f5cf7a700) :  INFO (agentkit:957) - Searching for matches for {'openSubtitlesHash': '23d18b3278dc08b7', 'name': 'Familystrokes 2019 12 26 1079 Christmas Family Orgy', 'filename': '%2Fprivate%2FFamilyStrokes%2FFamilyStrokes%20-%202019-12-26%20-%201079%20-%20Christmas%20Family%20Orgy%20[Charlotte%20Sins%2C%20Summer%20Hart]%2Emp4', 'plexHash': '70a2d6ed85ebf14cc1b09b735b29228f002e4e71', 'duration': '2275506', 'id': '52274'}
2019-12-27 03:55:38,560 (150f5cf7a700) :  DEBUG (networking:143) - Requesting 'http://127.0.0.1:32400/library/metadata/52274/tree'
2019-12-27 03:55:38,564 (150f5cf7a700) :  INFO (__init__:44) - *******MEDIA TITLE****** Familystrokes 2019 12 26 1079 Christmas Family Orgy
2019-12-27 03:55:38,564 (150f5cf7a700) :  INFO (__init__:51) - Getting Search Settings for: Familystrokes 2019 12 26 1079 Christmas Family Orgy
2019-12-27 03:55:38,564 (150f5cf7a700) :  INFO (PAsearchSites:963) - mediaTitle w/ possible abbrieviation: Familystrokes 2019 12 26 1079 Christmas Family Orgy
2019-12-27 03:55:38,574 (150f5cf7a700) :  INFO (PAsearchSites:1153) - mediaTitle w/ possible abbrieviation fixed: Familystrokes 2019 12 26 1079 Christmas Family Orgy
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1168) - ^^^^^^^ siteID: 626
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1170) - ^^^^^^^ Shortening Title
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1171) - familystrokes  vs family strokes
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1230) - 3
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1230) - 7
2019-12-27 03:55:38,577 (150f5cf7a700) :  INFO (PAsearchSites:1230) - searchTitle (before date processing): 2019 12 26 1079 Christmas Family Orgy
2019-12-27 03:55:38,578 (150f5cf7a700) :  INFO (__init__:61) - Search Title: 1079 Christmas Family Orgy
2019-12-27 03:55:38,578 (150f5cf7a700) :  INFO (__init__:65) - Search Date: 2019-12-26
2019-12-27 03:55:38,578 (150f5cf7a700) :  INFO (__init__:70) - 1079%20Christmas%20Family%20Orgy
2019-12-27 03:55:38,583 (150f5cf7a700) :  INFO (siteFamilyStrokes:8) - Title Search
2019-12-27 03:55:38,584 (150f5cf7a700) :  DEBUG (networking:143) - Requesting 'http://familystrokes.org/1079-christmas-family-orgy'
2019-12-27 03:55:39,280 (150f5cf7a700) :  INFO (siteFamilyStrokes:13) - Fan Page url: http://familystrokes.org/1079-christmas-family-orgy
2019-12-27 03:55:39,281 (150f5cf7a700) :  INFO (siteFamilyStrokes:16) - SceneTitle: Sexy Selfies For Her Stepgrandpa
2019-12-27 03:55:39,281 (150f5cf7a700) :  INFO (siteFamilyStrokes:34) - attempting direct url match
2019-12-27 03:55:39,281 (150f5cf7a700) :  INFO (siteFamilyStrokes:36) - SearchString: 1079-Christmas-Family-Orgy
2019-12-27 03:55:39,282 (150f5cf7a700) :  DEBUG (networking:143) - Requesting 'https://www.familystrokes.com/scenes/1079-Christmas-Family-Orgy'
2019-12-27 03:55:39,645 (150f5cf7a700) :  INFO (siteFamilyStrokes:39) - Source Page: https://www.familystrokes.com/scenes/1079-Christmas-Family-Orgy
2019-12-27 03:55:39,646 (150f5cf7a700) :  CRITICAL (agentkit:1014) - Exception in the search function of agent named 'PhoenixAdult', called with keyword arguments {'openSubtitlesHash': '23d18b3278dc08b7', 'name': 'Familystrokes 2019 12 26 1079 Christmas Family Orgy', 'filename': '%2Fprivate%2FFamilyStrokes%2FFamilyStrokes%20-%202019-12-26%20-%201079%20-%20Christmas%20Family%20Orgy%20[Charlotte%20Sins%2C%20Summer%20Hart]%2Emp4', 'plexHash': '70a2d6ed85ebf14cc1b09b735b29228f002e4e71', 'duration': '2275506', 'id': '52274'} (most recent call last):
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-ac2afe5f8/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/api/agentkit.py", line 1007, in _search
    agent.search(*f_args, **f_kwargs)
  File "/config/Library/Application Support/Plex Media Server/Plug-ins/PhoenixAdult.bundle/Contents/Code/__init__.py", line 704, in search
    results = PAsearchSites.siteFamilyStrokes.search(results, encodedTitle, title, searchTitle, siteNum, lang, searchByDateActor, searchDate, searchSiteID)
  File "/config/Library/Application Support/Plex Media Server/Plug-ins/PhoenixAdult.bundle/Contents/Code/siteFamilyStrokes.py", line 40, in search
    searchResult = searchResults.xpath('//div[@id="scenes-card"]')[0]
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-ac2afe5f8/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/code/sandbox.py", line 108, in <lambda>
    _getitem_           = lambda x, y: x.__getitem__(y),
IndexError: list index out of range

2019-12-27 03:55:39,647 (150f5cf7a700) :  DEBUG (runtime:88) - Sending packed state data (1044 bytes)
2019-12-27 03:55:39,648 (150f5cf7a700) :  DEBUG (runtime:924) - Response: [200] str, 16 bytes
```

**Comment by claygoldfinch** [comment]
FamilyStrokes is currently DirectURL rather than SceneID - you have to enter the entire "end" of the URL as the search term (ie. FamilyStrokes - 1079/1/Christmas-Family-Orgy). 
```
# revert to direct url match with official site webpage
    except:
        Log("attempting direct url match")
        searchString = searchTitle.replace(" ","-").replace(",","").replace("'","").replace("?","")
        Log("SearchString: " + searchString)
        fanUrl = PAsearchSites.getSearchSearchURL(searchSiteID) + searchString
        searchResults = HTML.ElementFromURL(fanUrl)
        Log('Source Page: ' + fanUrl)
        searchResult = searchResults.xpath('//div[@id="scenes-card"]')[0]
        titleNoFormatting = searchResult.xpath('.//div[contains(@class,"left-info")]//span')[0].text_content().strip()
        Log("Result Title: " + titleNoFormatting)
        curID = searchResults.xpath('.//link[@rel="canonical"]')[0].get('href').replace('/','_').replace('?','!')
        Log("curID: " + curID)
        releaseDate = parse(searchResult.xpath('.//div[@class="scene-date"]')[0].text_content().strip()).strftime('%Y-%m-%d')
        Log("releaseDate: " + releaseDate)
        girlName = searchResult.xpath('.//div[@class="starring"]//span')[0].text_content().strip()
        if searchDate:
            score = 100 - Util.LevenshteinDistance(searchDate, releaseDate)
        else:
            score = 95
```

It also utilizes Title Search via a fansite:
```
    try:
        Log("Title Search")
        fanSearchBase = "http://familystrokes.org/"
        title_searchString = searchTitle.lower().replace(" ","-").replace(",","").replace("'","")
        url = fanSearchBase + title_searchString
        title_searchResults = HTML.ElementFromURL(url)
        Log("Fan Page url: " + url)
        title_searchResult = title_searchResults.xpath('//div[@id="content-inside"]')[0]
        titleNoFormatting = title_searchResult.xpath('.//h2')[0].text_content().strip()
        Log("SceneTitle: " + titleNoFormatting)
        curID = title_searchResults.xpath('//link[@rel="canonical"]')[0].get('href').replace("/","_").replace("?","!")
        Log("curID: " + curID)
        try:
            releaseDate = parse(title_searchResult.xpath('.//div[@id="title-single"]//span[1]')[0].text_content().strip()).strftime('%Y-%m-%d')
            Log("ReleaseDate Parsed: " + releaseDate)
        except:
            releaseDate = ''
        girlName = title_searchResult.xpath('.//div[@id="title-single"]//a')[0].text_content()
        Log("firstActressName: " + girlName)
        if searchDate:
            score = 100 - Util.LevenshteinDistance(searchDate, releaseDate)
        else:
            score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
```

If you're up for it, you could certainly upgrade FamilyStrokes from DirectURL to SceneID - https://www.familystrokes.com/scenes/1079/1/1 works just as well as https://www.familystrokes.com/scenes/1079/1/Christmas-Family-Orgy

**Comment by HWatBobby** [comment]
It's weird because I haven't had an issue with matching the scenes with the way I've been naming the files until now and I just can't see any changes to the site. I did the fansite way and got a match, but no metadata came with it. And I did look into the 1079/1/1 method, but by changing the last string to a "1" I can see the scene name changes on the page to "1" and the scene name in the HTML is also changed to "1", so it wouldn't pick up the correct scene name. TeamSkeet makes things difficult with all of these different layouts across their sites. I guess I'll be going with your first suggestion so I can get the metadata with it until I can get this figured out.
